### PR TITLE
Profiling distributed search tree views

### PR DESCRIPTION
**Pull request by masterleinad**
Analogon to #381 for DistributedSearchTree. I still like to distinguish between nearest_queries and spatial_queries if we can but won't fight over it. Also, we could append `ArborX::` to `Testing` in both pull requests.

Output for `ArborX_DistributedTree.exe`:
```
ALLOCATIONS AT TIME OF HIGH WATER MARK:
  23.2% internal_and_leaf_nodes
  14.4% ArborX:BVH:spatial_queries/ArborX:BVH:two_pass/ArborX:BVH:two_pass:second_pass/Testing::values
  14.4% Testing::values
  8.7% Testing::bounding_boxes
  7.2% ArborX::DistributedSearchTree::forwardQueries::import_ranks
  7.2% ArborX::DistributedSearchTree::forwardQueries::import_ids
  7.2% ArborX::DistributedSearchTree::forwardQueries::import_ranks
  7.2% ArborX::DistributedSearchTree::forwardQueries::import_ids
  4.3% Testing::values
  2.3% ArborX::DistributedSearchTree::forwardQueries::queries
  1.7% Testing::queries
  0.6% ArborX:BVH:spatial_queries/ArborX:BVH:spatial_queries:init_and_alloc/Testing::offsets
  0.6% ArborX::DistributedSearchTree::forwardQueries::import_ids
  0.6% ArborX:BVH:spatial_queries/ArborX:BVH:two_pass/ArborX:BVH:two_pass:second_pass/ArborX::DistributedSearchTree::spatial_query::indices
  0.6% ArborX::DistributedSearchTree::forwardQueries::import_ranks
```

### CODE REVIEW

**Review by aprokop — CHANGES_REQUESTED** [review]
This goes beyond #380, also encompassing #381. #381 has not been yet decided on, thus this is blocked until #381 is merged. On the other hand, if the changes from `src/` are removed, then this can be merged.

**Comment by masterleinad** [comment]
Rebased after #381 has been merged. I am happy to change the style again if required.

File: benchmarks/distributed_tree_driver/distributed_tree_driver.cpp
Comment (aprokop) [inline benchmarks/distributed_tree_driver/distributed_tree_driver.cpp:403]: Please note that in #389 we also updated the style of profiling regions.

File: src/ArborX_DistributedSearchTree.hpp
Comment (aprokop) [inline src/ArborX_DistributedSearchTree.hpp:143]: One of the thing that @dalg24 identified when working on the related PRs is that clang-format sometimes breaks in inopportune places. In situations like this one, it would be better to break after '::' by manually creating a new line ourselves.

File: src/ArborX_DistributedSearchTree.hpp
Comment (aprokop) [inline src/ArborX_DistributedSearchTree.hpp:159]: See comment above.

File: src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp
Comment (aprokop) [inline src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp:83]: `spatial_queries` -> `query::spatial`

**Review by aprokop — CHANGES_REQUESTED** [review]
From the code perspective, this is mostly fine (see comments). But this also needs a test to make sure the guidelines are followed. Please see how #dalg24 added testing for `BVH`.

**Comment by masterleinad** [comment]
Rebased and hopefully addressed all comments.

File: test/tstDistributedKokkosToolsAnnotations.cpp
Comment (dalg24) [inline test/tstDistributedKokkosToolsAnnotations.cpp:71]: These corner case make sense on the BVH because they explore different code path.  I am not sure what they bring to the table for the distributed tree.

Comment (masterleinad) [reply test/tstDistributedKokkosToolsAnnotations.cpp:71]: I think it makes sense to have at least the empty case and something else but I am happy to drop the empty case as well if you are happier with that.

Comment (aprokop) [reply test/tstDistributedKokkosToolsAnnotations.cpp:71]: Agreed. Leave empty and remove one and two leaves cases.

**Comment by aprokop** [comment]
This needs rebasing, as I changed the name of the distributed annotations test file in #384.

**Comment by masterleinad** [comment]
Rebased.

**Comment by masterleinad** [comment]
I am happy to squash everything into one commit if preferred.

File: test/tstKokkosToolsDistributedAnnotations.cpp
Comment (aprokop) [inline test/tstKokkosToolsDistributedAnnotations.cpp:62]: I'm confused about the purpose of this function. What does it test that the next one does not?

File: src/details/ArborX_DetailsTreeTraversal.hpp
Comment (aprokop) [inline src/details/ArborX_DetailsTreeTraversal.hpp:195]: I think it would be fine to leave this one alone. If you do decide to change it, you would also need to change other places in the file, for example, spatial queries.

But, I do think this file needs a more thorough cleanup. @dalg24 I did not notice at the time, but `TreeTraversal` file has a mix of `TreeTraversal` and `BVH` labels. I want to confirm that it was intentional, and suggest changing all to `TreeTraversal` instead of `BVH`. But it does not have to be done now.

**Review by aprokop — COMMENTED** [review]
I think it's almost there.

File: src/details/ArborX_DetailsTreeTraversal.hpp
Comment (dalg24) [reply src/details/ArborX_DetailsTreeTraversal.hpp:195]: Oversight on my part.  Sounded like a good idea at the time.

File: test/tstKokkosToolsDistributedAnnotations.cpp
Comment (masterleinad) [reply test/tstKokkosToolsDistributedAnnotations.cpp:62]: We are checking the labels for the allocations in the construction here instead of the labels for the allocations when performing the search.

**Review by aprokop — APPROVED** [review]
Looks good.

**Comment by aprokop** [comment]
Merged with the usual HIP failure.

File: src/details/ArborX_DetailsTreeTraversal.hpp
Comment (aprokop) [reply src/details/ArborX_DetailsTreeTraversal.hpp]: Resolved in #398.

**Pull request merged**
